third-party/qthread/Makefile

```diff
@@ -86,7 +86,9 @@ ifeq ($(CFLAGS_NEEDS_RT_INCLUDES), y)
 endif
 
 # enable oversubscription for testing
-ifneq (, $(CHPL_QTHREAD_ENABLE_OVERSUBSCRIPTION))
+ifneq (, $(call isTrue, $(CHPL_RT_OVERSUBSCRIBED)))
```

Comment:
I'm not sure we want this to be tied to `CHPL_RT_OVERSUBSCRIBED` (I interpret `CHPL_RT_` vars to be things that control run time, not build time behavior.) I've also been wondering if we can get away with not using `--enable-oversubscription` anymore. I don't love that it's impacts the build, but we don't include it in the build path and I'm not sure how much it actually helps. As a for instance we don't set this for paratesting, we only set `QT_AFFINITY=no` and `QT_SPINCOUNT=300` and we run paratests heavily oversubscribed without timeouts. i.e. I think affinity and spinwaiting are the big oversubscription killers, so I'm wondering if we can just use those and remove this option.

 `--enable-oversubscription` option does 3 things:
 - lowers the default `QT_SPINCOUNT` from 300,000 to 300 (we'll be doing this ourselves now)
 - turns atomic spinwait locks into pthread locks 
 - has busy waits do sched_yield

I'm not sure how much using pthread locks and sched_yield actually helps in an oversubscribed situation. I think I'd like to run some experiments on low and high core machines to get a sense of whether this option is worth keeping around, but regardless I don't think we want to tie it to `CHPL_RT_OVERSUBSCRIBED`

Reply:
Here's some quick timings that I think demonstrate that `--enable-oversubscription` doesn't really help us (it helps some on 16-64 locales for chapcs, but beyond that it doesn't seem to help much, and it has very little impact on lower core-count machines):

`chpl examples/hello4-datapar-dist.chpl` w/ gasnet-udp opt/non-debug runtime:

"default" env vars:
```sh
export GASNET_SPAWNFN=L
export GASNET_ROUTE_OUTPUT=0
export CHPL_COMM=gasnet
export QT_SPINCOUNT=300
export QT_AFFINITY=no
export GASNET_QUIET=1
```

additional "oversub" env var:
```sh
export CHPL_QTHREAD_ENABLE_OVERSUBSCRIPTION=1
# clobber/rebuild qthreads
```

chapcs (24 cores, 48 threads):

| locales | default | oversub |
| ------- | ------- | ------- |
|   1     |  0.322s |  0.322s |
|   2     |  0.458s |  0.463s |
|   4     |  0.534s |  0.523s |
|   8     |  0.845s |  0.655s |
|  16     |  1.610s |  1.031s |
|  32     |  3.281s |  2.067s |
|  64     |  7.301s |  4.704s |
| 128     | 17.879s | 16.936s |
| 256     | 56.457s | 53.795s |

chapvm (4 cores, 4 threads):

| locales | default | oversub |
| ------- | ------- | ------- |
|   1     |  0.315s |  0.313s |
|   2     |  0.969s |  0.777s |
|   4     |  2.540s |  2.415s |
|   8     |  6.649s |  7.044s |
|  16     | 17.108s | 18.562s |
|  32     | 37.530s | 38.538s |

my mac (2 cores, 4 threads):

| locales | default | oversub |
| ------- | ------- | ------- |
|   1     |  0.025s |  0.038s |
|   2     |  0.399s |  0.360s |
|   4     | 16.744s | 16.521s |
|   8     | 43.545s | 47.432s |